src/backend/src/lib.rs

```diff
@@ -165,6 +165,14 @@ fn post_upgrade(arg: Option<Arg>) {
     }
 }
 
+/// Show the canister configuration.
+#[query(guard = "caller_is_allowed")]
+#[allow(clippy::needless_pass_by_value)]
+#[must_use]
+fn config() -> Config {
```

Comment:
Can you provide tests? Like one that assert the result and one that checks that the function can only be called allowed callers?

Reply:
Sure thing.  The tests are a lot bigger than this change so I was going to put them in a separate PR but I can include them here.  Done.